### PR TITLE
ZALLY-92 Add Initial Integration Tests Suite

### LINKED ISSUE #92: Add an Integration Test to Run on Travis.CI
In order to make sure that CLI tool is not broken, we can add a small integration test which:

1. Compiles and starts Zally Server
2. Compiles Zally CLI
3. Runs few scenarios with local and remote files

In order to keep it simple, we can use bats framework for that.

### PR DESCRIPTION
**Pull request by netme**
Resolves #92 

- [x] Adds simple `.bats` script to run CLI tool agains locally running server. 
- [x] Simplifies usage of `shadowJar`

More info: https://github.com/sstephenson/bats

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,9 +5,37 @@ language: java
...
+
+  # Launch local zally server
+  - cd server/
+  - echo "spring.profiles.active=dev" > application.properties
+  - ./gradlew bootRun > /dev/null &
+  - echo $! > /tmp/zally_server.pid
+  - cd ..
+
+  # Wait until Spring Boot will start
+  - sleep 15
```
Comment (netme) [inline .travis.yml]: We have to wait until https://github.com/spring-projects/spring-boot/issues/3982 will be resolved :-)

File: .travis.yml [hunk 2]
```diff
@@ -5,9 +5,37 @@ language: java
...
 install: true
+
 script:
+  # Unit-test and build CLI tool
   - cd cli/
-  - gradle test --info
-  - cd ../server/
-  - gradle test --info
+  - ./gradlew test --info
+  - ./gradlew assemble
```
Comment (fmueller) [inline .travis.yml]: I propose to use `./gradlew build` instead of test and assemble afterwards. build is executing all tests and builds the jar file.

File: cli/README.md [hunk 3]
```diff
@@ -14,7 +14,7 @@ requesting violations check at a given Zally server.
 
 3. Build `zally.jar` package:
 	```bash
-	./gradlew shadowJar
+	./gradlew assemble
```
Comment (fmueller) [inline cli/README.md]: As comment above: build instead of assemble.

File: cli/build.gradle [hunk 4]
```diff
@@ -45,3 +45,5 @@ shadowJar {
 task wrapper(type: Wrapper) {
     gradleVersion = '2.14.1'
 }
+
+assemble.dependsOn shadowJar
```
Comment (fmueller) [inline cli/build.gradle]: I don't know if this is still necessary if we stick to build instead of assemble.

File: .travis.yml (hunk 2 above)
Comment (netme) [inline .travis.yml]: @fmueller thanks for your comment, I will investigate a bit more on this topic. Also was not sure if `assemble` is the best option.

File: cli/build.gradle (hunk 4 above)
Comment (netme) [inline cli/build.gradle]: `shadowJar` is not triggered by `build`. That's why I've started doing that changes. I will try to find a better solution.

Comment (fmueller) [inline cli/build.gradle]: Adding shadowJar step to `build`, maybe?

Comment (netme) [inline cli/build.gradle]: Yeah, most probably it will work. I'm still quite new to `gradle`, sorry :-)

Comment (fmueller) [inline cli/build.gradle]: Don't wanted to sound rude. I like your contributions a lot. ;-)

Comment (netme) [inline cli/build.gradle]: You are not sounding rude, I really appreciate your feedback :-)

**Comment by netme** [comment]
@fmueller I've switched `assemble` to `build` in 1357455 .

File: .travis.yml [hunk 5]
```diff
@@ -5,9 +5,37 @@ language: java
...
+
 install: true
+
 script:
+  # Unit-test and build CLI tool
   - cd cli/
-  - gradle test --info
-  - cd ../server/
-  - gradle test --info
+  - ./gradlew test --info
```
Comment (fmueller) [inline .travis.yml]: With `build` this test goal should also be executed. I think, we can remove `gradle test` here.

Comment (netme) [inline .travis.yml]: Good point, let's modify the travis script and check it :-)

Comment (netme) [inline .travis.yml]: @fmueller Fixed

Comment (netme) [inline .travis.yml]: And many thanks for your comments, I really learned a lot about Gradle these days 👍

**Comment by maxim-tschumak** [comment]
:+1: from my side. Bats looks very good! I would go for fewer  integration tests (one happy case, one not-very-happy case). It's easier to maintain and other corner cases should be tested on unit level, IMO.

**Comment by netme** [comment]
@maxim-tschumak Thanks for checking! I agree, the reason why we add 5 of them is to cover all 4 happy-paths and one corner case. We found in the past that it's easy to break one of the possible flows in CLI tool. Later, if you don't mind, I would like to add the tests for another service URL and token. But let's discuss it in person. 

@fmueller, your approval is still missing :-)
